packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js

```diff
@@ -16,13 +16,16 @@ import useInitEditedEntityFromURL from '../sync-state-with-url/use-init-edited-e
 import usePatternDetails from './use-pattern-details';
 import { store as editSiteStore } from '../../store';
 import { unlock } from '../../lock-unlock';
+import normalizePostIdForPostType from '../../utils/normalize-post-id-for-post-type';
 
 export default function SidebarNavigationScreenPattern() {
-	const { params } = useNavigator();
 	const { categoryType } = getQueryArgs( window.location.href );
-	const { postType, postId } = params;
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
 
+	const { params } = useNavigator();
+	const { postType } = params;
+	const postId = normalizePostIdForPostType( params?.postId, postType );
```

Comment:
Patterns can be `wp_block` or `wp_template_part` which use different types for recordKey so we need to selectively normalize.